pkg/apis/core/validation/validation.go

```diff
@@ -4688,6 +4688,11 @@ func ValidatePodEphemeralContainersUpdate(newPod, oldPod *core.Pod, opts PodVali
 	allErrs = append(allErrs, validatePodMetadataAndSpec(newPod, opts)...)
 	allErrs = append(allErrs, ValidatePodSpecificAnnotationUpdates(newPod, oldPod, fldPath.Child("annotations"), opts)...)
 
+	// static pods don't support ephemeral containers #113935
+	if _, ok := oldPod.Annotations[core.MirrorPodAnnotationKey]; ok {
+		return field.ErrorList{field.Forbidden(field.NewPath(""), fmt.Sprintf("static pods (%q) do not support ephemeral containers", oldPod.ObjectMeta.GetName()))}
```

Comment:
shouldn't need to include the name here, returned errors should get wrapped in an overall error that includes the name